core/queueref.go

```diff
@@ -1,9 +1,21 @@
 package core
 
 import (
+	"errors"
+	"sync"
 	"unsafe"
 )
 
+var (
+	MaxQueueSize = 10000
+	MaxQueues    = 1000
+)
+
+var (
+	QueueCount = 0
+	muQueue    sync.Mutex
```

Comment:
Why do we require a mutex?

Reply:
Since it's a global variable, multiple clients trying to create new stacks concurrently will lead to thread safety issues.

Reply:
If these values are not being mutated there will not be any thread safety issues.
We can declare these variables as constants instead to guarantee immutability.

Reply:
we will increment it when new stack is created.

Reply:
But obvious, so silly of me! Can we then just lock the critical part then?
```suggestion
	muQueue.Lock()
	QueueCount++
	muQueue.Unlock()	
```

Reply:
yeah sure.

Reply:
Done